src/Native/Unix/Common/pal_utilities.h

```diff
@@ -16,6 +16,28 @@
 #include <type_traits>
 #include <unistd.h>
 
+#ifdef DEBUG
+#define assert_err(cond, msg, err) do \
+{ \
+  if(!(cond)) \
+  { \
+    fprintf(stderr, "%s (%d): error %d: %s. %s\n", __FILE__, __LINE__, err, msg, strerror(err)); \
+    assert(cond && "assert_err failed"); \
+  } \
+} while(0)
+#define assert_msg(cond, msg, val) do \
+{ \
+  if(!(cond)) \
+  { \
+    fprintf(stderr, "%s (%d): error %d: %s\n", __FILE__, __LINE__, val, msg); \
```

Comment:
>  the asserts does represent an working

Sure, all asserts do. But in the places this is used, the value isn't an error, rather an arbitrary input that was associated with the condition somehow. It's strange then to have "error 42" when 42 isn't an error.  I'm not clear on why matching the msbuild format is so important, but if we must, I think it'd be better to use 0 or -1 for the error code or something else more likely to be seen as a sentinel, and then include the input as part of the formatted message portion of the line.